### DISCUSSION TITLE
Uf gate in Deutsch's algorithm

### DISCUSSION THREAD

**Discussion by Da-artof** [Qiskit/.github#35] [General Feedback]
Hello, I am trying to implement Deutsch's algorithm in Qiskit, but am struggling as to how to create the Uf gate (query gate) for a given single qubit function f as given in this diagram:

<img width="629" alt="Screenshot 2023-08-01 at 16 11 19" src="https://github.com/Qiskit/feedback/assets/85499672/eda49bb4-37a1-4070-a008-f47a768a1df0">

from the Qiskit Lesson: https://www.youtube.com/watch?v=2wticzHE1vs at 25:00.
any help would be appreciated, thanks.

**Comment by 1ucian0** [comment]
Take a look to https://github.com/qiskit-community/qiskit-presentations/blob/master/2022-06-13_Oxford-IBM_Alliance/intro_QC_with_Qiskit.ipynb (the Deutsch section).
